server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -541,6 +557,54 @@ private ClusterState applyCreateIndexWithTemporaryService(
         });
     }
 
+    Template applyContext(
+        CreateIndexClusterStateUpdateRequest request,
+        ClusterState currentState,
+        List<Map<String, Object>> mappings,
+        Settings.Builder settingsBuilder
+    ) throws IOException {
+        if (request.context() != null) {
+            ComponentTemplate componentTemplate = MetadataIndexTemplateService.findComponentTemplate(
+                currentState.metadata(),
+                request.context()
+            );
+
+            if (componentTemplate.template().mappings() != null) {
+                // Mappings added at last (priority to mappings provided)
+                mappings.add(MapperService.parseMapping(xContentRegistry, componentTemplate.template().mappings().toString()));
+            }
+
+            if (componentTemplate.template().settings() != null) {
+                validateOverlap(settingsBuilder.keys(), componentTemplate.template().settings(), request.index()).ifPresent(message -> {
+                    ValidationException validationException = new ValidationException();
+                    validationException.addValidationError(message);
```

Comment:
In case a new setting got added to context template which is conflicting, then what would happen to
1. Auto index creation via component template
2. Explicit index creation via API
would validation error be thrown in both the cases. Case 1 could be a concern for user where everything was working and things would start failing all of a sudden

Reply:
For auto-create, no it will not cause failures since in those cases, context is tied to index template where context is given the highest priority while ignoring other declared settings. This ensures that existing flows around rollover, auto-create don't break.

For explicit index creation, the error tells users that when they are declaring context, it will fail and they can update the request.

This ensures that we maintain backward compatibility in cases new settings are added to context for applicable cases. In case we foresee that silent ignoring may be a better idea, we can change the flow since existing context based indices will not break